src/store/utils.js

```diff
@@ -1,16 +1,21 @@
 
+import APIError from '../utils/api/errors'
+
 // factories
 export function doRequestFactory(REQUEST, COMMITS) {
   return async function ({ commit }, { action, args, load = 'TOGGLE_LOADING' }) {
     if (load) commit(load, true)
+    let ret = new APIError('Success', 0, 200)
     try {
       const data = await REQUEST[action](...args)
       commit(COMMITS[action], data)
     } catch (e) {
       commit('API_ERROR', e)
+      ret = e
```

Comment:
hmmmm

Reply:
The idea behind this is that the view/component/whatever can just inspect the return value now. 
Previously, you had to do the dispatch, then check the error value in the store, which would obviously break or be harder to reason about in the presence of multiple dispatches.